hail/python/test/hail/methods/test_statgen.py

```diff
@@ -476,24 +476,24 @@ def test_weighted_linear_regression(self):
         mt = hl.import_vcf(resource('regressionLinear.vcf'))
         mt = mt.add_col_index()
 
-        mt = mt.annotate_cols(y=hl.coalesce(pheno[mt.s].Pheno, 1.0))
+        mt = mt.annotate_cols(y=hl.coalesce(pheno[mt.s].Pheno, 1.0), weight=(mt.col_idx % 2) + 1, weight2=3 * (mt.col_idx % 3) + 1)
         mt = mt.annotate_entries(x=hl.coalesce(mt.GT.n_alt_alleles(), 1.0))
         my_covs = [1.0] + list(covariates[mt.s].values())
 
         ht_with_weights = hl._linear_regression_rows_nd(y=mt.y,
                                           x=mt.x,
                                           covariates=my_covs,
-                                          weights=mt.col_idx)
+                                          weights=mt.weight)
```

Comment:
I was getting test failures with the old weights. I think what was happening is that using col_idx as a weight created a very badly conditioned regression problem, so different math that's equivalent in exact arithmetic could lead to substantially different results in floating point.